// These messages are specific to the Open Source API, outside of the Public
// Cloud Vizier API.

syntax = "proto3";

package vizier;

import "google/protobuf/timestamp.proto";

// The EarlyStoppingOperation will be used internally by the Vizier service to
// to represent the state of a computation which determines if a Trial should
// stop early.
message EarlyStoppingOperation {
  string name = 1;

  enum Status {
    UNKNOWN = 0;
    // An EarlyStoppingOperation that is ACTIVE is being processed.
    ACTIVE = 1;
    // The EarlyStoppingOperation has completed. In this case, should_stop
    // will be populated.
    DONE = 2;
    // The EarlyStoppingOperation could not be completed. If status == FAILED,
    // failure_code and failure_message will be set.
    FAILED = 3;
  }
  // REQUIRED
  Status status = 2;
  // If true, the Trial should stop. Only set if status==DONE.
  bool should_stop = 3;
  // If the operation has a status of FAILED, a failure code and user readable
  // message will be populated here.
  string failure_message = 4;

  // Time the EarlyStoppingOperation was first created.
  google.protobuf.Timestamp creation_time = 5;

  // Time the EarlyStoppingOperation was completed.
  google.protobuf.Timestamp completion_time = 6;
}
